pyOCD/pyDAPAccess/dap_access_cmsis_dap.py

```diff
@@ -165,46 +164,77 @@ def _dap_transfer_block_free_bytes(self, size):
         recv = size - 4 - 4 * self._read_count
         return send, recv
 
-    def get_read_space(self):
+    def _get_free_read_transfer_words(self):
         """
         Return the number of words free in the response packet
         """
-        # Size left if useing DAP_Transfer
         send, recv = self._dap_transfer_free_bytes(self._size)
-        transfer_size = min(send, recv // 4)
-
-        # Size left if using DAP_TransferBlock
-        _, recv = self._dap_transfer_block_free_bytes(self._size)
-        transfer_block_size = recv // 4
-
-        # If data does not allow block transfer invalidate it
-        if not self._block_allowed:
-            transfer_block_size = -1
-
-        size = max(transfer_size, transfer_block_size)
-        assert size >= 0
-        return size
+        return min(send, recv // 4)
 
-    def get_write_space(self):
+    def _get_free_write_transfer_words(self):
         """
         Return the number of words free in the transmit packet
         """
-        # Size left if useing DAP_Transfer
         send, _ = self._dap_transfer_free_bytes(self._size)
-        transfer_size = send // 5
+        return send // 5
 
-        # Size left if using DAP_TransferBlock
+    def _get_free_read_transfer_block_words(self):
+        """
+        Return the number of words free in the response packet
+        """
+        _, recv = self._dap_transfer_block_free_bytes(self._size)
+        return recv // 4
+
+    def _get_free_write_transfer_block_words(self):
+        """
+        Return the number of words free in the transmit packet
+        """
         send, _ = self._dap_transfer_block_free_bytes(self._size)
-        transfer_block_size = (self._size - 5) // 4 - self._write_count
+        return send // 4
 
-        # If data does not allow block transfer invalidate it
-        if not self._block_allowed:
-            transfer_block_size = -1
+    def get_request_space(self, count, request, dap_index):
```

Comment:
Ah, makes sense. I wasn't looking closely enough at the code. Yeah, that logging looks like a good feature.